lib/api/signal.js

```diff
@@ -1,24 +1,51 @@
-export default class Signal {
+export class Signal {
   constructor () {
     this._id = 0
     this._listeners = {}
   }
 
   dispatch (...args) {
     for (let key in this._listeners) {
-      this._listeners[key](...args)
+      this._listeners[key].apply(null, args)
     }
   }
 
   attach (listener) {
     if (typeof listener !== 'function') {
       throw new Error('listener function expected')
     }
-    var id = this._id++
+    const id = this._id++
     this._listeners[id] = listener
-    var self = this
-    return function removeListener () {
-      delete self._listeners[id]
+    // return function that'll detach the listener
+    return () => delete this._listeners[id]
+  }
+}
+
+const memArgsSymbol = Symbol('Private memoized arguments')
```

Comment:
[Not supported](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) by IE :(